lib/rspec_profiling/example.rb

```
@@ -79,9 +80,16 @@ def log_request(request, start, finish)
       counts[:request_time] += request[:view_runtime].to_f
     end
 
-    def log_event(name, start, finish)
-      event_counts[name] += 1
-      event_times[name] += (finish - start)
+    def log_event(event_name, event, start, finish)
+      event_counts[event_name] += 1
+      event_times[event_name] += (finish - start)
+      if verbose_record_event?(event_name)
+        begin
+          ((event_events[event_name] ||= []) << event.as_json)
+        rescue => e
```

I don't love silently swallowing these. We've indicated that we want to record them, so at this point we at least _think_ they are serializable.

For later - I wouldn't hate the ability to supply a custom formatter:

```ruby
it 'records', record_events: ['deprecation.rails', 'factory_bot' => FactoryBotEventFormatter] { ... }
```

It's _possible_ we should just be storing the `event` object here, and the casting to JSON or whatever should be a concern for the storage adapter. Want to pick that question up when we work on the PSQL adapter?

---

I really don't enjoy the no-op either and 100% agree on the custom formatter and called that out in the main PR. I _believe_ the issue I'm running into is because of our class serialize problem we've seen elsewhere. So this was a stop gap for now so tests don't fail because of a possible failure of our json building.